core/domain/exp_services.py

```diff
@@ -1188,7 +1322,12 @@ def update_exploration(
             'Commit messages for non-suggestions may not start with \'%s\'' %
             feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX)
 
-    exploration = apply_change_list(exploration_id, change_list)
+    exploration_before_change = get_exploration_by_id(exploration_id)
+    exploration_after_change = apply_change_list(exploration_id, change_list)
```

Comment:
@seanlip & @nithusha21, will it be fine to send exploration as a parameter to `apply_change_list`? (we are calling `get_exploration_by_id` twice, above this line as well as inside the `apply_change_list` method.)

Reply:
Better to use exploration_id and pass around strings where possible. It's fine to call get_by_id multiple times in succession as NDB datastore actually implements an internal cache that makes subsequent GET operations faster.